pwndbg/commands/heap.py

```diff
@@ -592,13 +592,19 @@ def find_fake_fast(addr, size=None, align=False):
     "max display size for heap chunks visualization (0 for display all)",
 )
 
+pwndbg.gdblib.config.add_param(
+    "default-visualize-chunk-number",
+    10,
+    "the number of chunks to visualize (default is 10)",
+)
+
 parser = argparse.ArgumentParser()
 parser.description = "Visualize chunks on a heap, default to the current arena's active heap."
 parser.add_argument(
     "count",
     nargs="?",
     type=lambda n: max(int(n, 0), 1),
-    default=10,
+    default=pwndbg.gdblib.config.default_visualize_chunk_number,
```

Comment:
The question is: does this really work? I think that maybe this is evaluated once when we add parser argument and is never "read again" for the command execution?

I think the test below may not test for the thing properly (see my comment there)

Reply:
> The question is: does this really work?

Yep, it works, it will be evaluated every time we execute `vis_heap_chunks`, the default value seems won't evaluate to int when initializing the parser.

```
pwndbg> pi pwndbg.commands.heap.vis_heap_chunks.parser._optionals._actions[1].default
<pwndbg.lib.config.Parameter object at 0x7f7310cc8df0>
pwndbg> set default-visualize-chunk-number 1
Set the number of chunks to visualize (default is 10) to 1
pwndbg> # if it's only evaluated once, `vis_heap_chunks` should still print out 10 chunks
pwndbg> vis_heap_chunks

0x55a4e3e76000	0x0000000000000000	0x0000000000000291	................
0x55a4e3e76010	0x0000000000000000	0x0000000100000000	................
0x55a4e3e76020	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76030	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76040	0x0000000000000000	0x0000000000000001	................
0x55a4e3e76050	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76060	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76070	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76080	0x0000000000000000	0x0001000000000000	................
0x55a4e3e76090	0x0000000000000000	0x0000000000000000	................
0x55a4e3e760a0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e760b0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e760c0	0x000055a4e3e76480	0x0000000000000000	.d...U..........
0x55a4e3e760d0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e760e0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e760f0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76100	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76110	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76120	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76130	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76140	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76150	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76160	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76170	0x000055a4e3e762a0	0x0000000000000000	.b...U..........
0x55a4e3e76180	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76190	0x0000000000000000	0x0000000000000000	................
0x55a4e3e761a0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e761b0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e761c0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e761d0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e761e0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e761f0	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76200	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76210	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76220	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76230	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76240	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76250	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76260	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76270	0x0000000000000000	0x0000000000000000	................
0x55a4e3e76280	0x0000000000000000	0x000055a4e3e76500	.........e...U..
0x55a4e3e76290	0x0000000000000000
pwndbg>
```